include/aos/common/tools/fs.hpp

```diff
@@ -348,6 +348,38 @@ class FS {
 #endif
         return ErrorEnum::eNone;
     }
+
+    /**
+     * Reads content of the file named by fileName into the given string.
+     *
+     * @param fileName file name.
+     * @param[out] text result string.
+     * @return Error.
+     */
+    static Error ReadFile(const String& fileName, String& text)
```

Comment:
Your C++ example more related how stream handle lines in text file rather than deal with string. How do you read whole file into std::string with one method in C++?

Also here is golang API:
```go
func ReadFile(name string) []byte
``` 

My consern here, you can't use this API for reading binary files in case we need it in the future. And we will need to implement separate methods for such case. Having byte array covers both cases and we may use one implementation for both.

But if you really like to read direct to string, then these methods should be renamed to: `ReadFileToString` and `WriteStringToFile` accordingly.